src/components/organism/CreatePool/index.tsx

```diff
@@ -0,0 +1,338 @@
+    }
+  };
+
+  const setSelectedTokenBValue = (value: number) => {
+    if (selectedTokenAssetValue && slippageValue) {
+      const assetTokenSlippageValue = calculateSlippageReduce(value, slippageValue);
+      const tokenWithSlippageFormatted = formatInputTokenValue(assetTokenSlippageValue, selectedTokenB?.decimals);
+      setSelectedTokenAssetValue({ tokenValue: value });
+      setAssetTokenWithSlippage({ tokenValue: parseInt(tokenWithSlippageFormatted) });
+    }
+  };
+
+  const checkAssetTokenMinAmount = async () => {
+    if (selectedTokenAssetValue && api && selectedTokenB.assetTokenId) {
+      const assetTokenInfo: any = await api.query.assets.asset(selectedTokenB.assetTokenId);
+      const assetTokenMinBalance = assetTokenInfo.toHuman()?.minBalance;
+      if (selectedTokenAssetValue.tokenValue >= assetTokenMinBalance?.replace(/[, ]/g, "")) {
+        setAssetTokenMinValueExceeded(false);
+      } else {
+        setAssetTokenMinValue(assetTokenMinBalance);
+        setAssetTokenMinValueExceeded(true);
+      }
+    }
+  };
+
+  const returnButtonStatus = () => {
+    if (!selectedTokenA.nativeTokenSymbol || !selectedTokenB.assetTokenId) {
+      return t("button.selectToken");
+    }
+    if (selectedTokenNativeValue?.tokenValue <= 0 || selectedTokenAssetValue?.tokenValue <= 0) {
+      return t("button.enterAmount");
+    }
+    if (selectedTokenNativeValue?.tokenValue > Number(tokenBalances?.balance)) {
+      return t("button.insufficientTokenAmount", { token: selectedTokenA.nativeTokenSymbol });
+    }
+    if (selectedTokenNativeValue?.tokenValue + parseFloat(poolGasFee) / 1000 > Number(tokenBalances?.balance)) {
+      return t("button.insufficientTokenAmount", { token: selectedTokenA.nativeTokenSymbol });
+    }
+    if (
+      selectedTokenAssetValue?.tokenValue >
+      formatDecimalsFromToken(parseInt(selectedTokenB.assetTokenBalance?.replace(/[, ]/g, "")), selectedTokenB.decimals)
+    ) {
+      return t("button.insufficientTokenAmount", { token: selectedTokenB.tokenSymbol });
+    }
+    if (selectedTokenA && selectedTokenB && poolExists) {
+      return t("button.enterAmount");
+    }
+    if (selectedTokenA && selectedTokenB) {
+      return assetTokenMinValueExceeded ? t("button.minimumTokenAmountExceeded") : t("button.deposit");
+    }
+  };
+
+  useEffect(() => {
+    if (tokenBalances) {
+      setSelectedTokenA({
+        nativeTokenSymbol: tokenBalances?.tokenSymbol as NativeTokenProps,
+        nativeTokenDecimals: tokenBalances?.tokenDecimals as NativeTokenProps,
+      });
+    }
+  }, [tokenBalances]);
+
+  useEffect(() => {
+    checkIfPoolAlreadyExists(selectedTokenB.assetTokenId);
+  }, [selectedTokenB.assetTokenId]);
+
+  useEffect(() => {
+    if (!poolExists && selectedTokenB.assetTokenId) {
+      handlePoolGasFee();
+    }
+  }, [poolExists, selectedTokenB.assetTokenId]);
+
+  useEffect(() => {
+    if (successModalOpen) setIsSuccessModalOpen(true);
```

Comment:
Why do u need this? U are asked if successModalOpen and then set it to true. If it's already true u do not need to set it again. Also, I do not get why u need this logic inside of the the useEffect. If u want to open a modal, just setIsSuccessModalOpen(true) on a specific action